server/services/star.js

```diff
@@ -51,14 +64,14 @@ module.exports = class StarService extends EventEmitter {
         homeStar.ships = homeStar.shipsActual;
         homeStar.naturalResources = game.constants.star.resources.maxNaturalResources; // Home stars should always get max resources.
         homeStar.homeStar = true;
-        homeStar.warpGate = false;
+        homeStar.warpGate = homeStar.warpGate ?? false;
 
         this.resetIgnoreBulkUpgradeStatuses(homeStar);
         
         // ONLY the home star gets the starting infrastructure.
-        homeStar.infrastructure.economy = gameSettings.player.startingInfrastructure.economy;
-        homeStar.infrastructure.industry = gameSettings.player.startingInfrastructure.industry;
-        homeStar.infrastructure.science = gameSettings.player.startingInfrastructure.science;
+        homeStar.infrastructure.economy = homeStar.infrastructure.economy ?? gameSettings.player.startingInfrastructure.economy;
+        homeStar.infrastructure.industry = homeStar.infrastructure.industry ?? gameSettings.player.startingInfrastructure.industry;
+        homeStar.infrastructure.science = homeStar.infrastructure.science ?? gameSettings.player.startingInfrastructure.science;
```

Comment:
We need to be mindful that there might be other game settings that could interfere with custom galaxies, have you checked other settings besides the starting infrastructure?